engine/src/flutter/impeller/entity/shaders/filters/gaussian.frag

```diff
@@ -40,5 +46,9 @@ void main() {
                           v_texture_coords + kernel_samples.sample_data[i].xy);
   }
 
-  frag_color = total_color;
+  if (frag_info.unpremultiply > 0.5) {
```

Comment:
Alternative option: Split the unpremultiply shader into another pipeline. However, I think the current implementation only adds a branching per pixel to `gaussian.frag`, which should be acceptable.